### DISCUSSION TITLE
How to install dependency in /src

### DISCUSSION THREAD

**Discussion by vodkaslime** [ether/etherpad-lite#6126] [Q&A] [answered]
Hi folks, wondering how to just install a dependency library in `/src` in etherpad-lite.

My current working environment is Ubuntu with node@21.1.0 and npm@10.2.0.

Let's say I want to hack and play a bit with etherpad-lite, so I intend to install `marked` to /src. Typically to install an npm package, we just do `npm install marked`, right? However, after installing with this, the `package.json.lock` is then upgraded to version 3, which cannot be verified by ehterpad-lite installation, as mentioned at https://github.com/ether/etherpad-lite/issues/6072.

So I manually reverted my working environment back to npm@5, but encountered more problems: `marked` requires higher version of node, rather than the ancient version of node@10 shipped with npm@5, etc. And also, it's obviously not best practice to always revert back to npm@5 to just please the lockfile version 1, right?

So getting a bit puzzled and wondering what is the best practice to add dependency to `/src`?

Thanks!

**Comment by SamTV12345** [comment, marked as answer]
> Hi folks, wondering how to just install a dependency library in `/src` in etherpad-lite.
> 
> My current working environment is Ubuntu with node@21.1.0 and npm@10.2.0.
> 
> Let's say I want to hack and play a bit with etherpad-lite, so I intend to install `marked` to /src. Typically to install an npm package, we just do `npm install marked`, right? However, after installing with this, the `package.json.lock` is then upgraded to version 3, which cannot be verified by ehterpad-lite installation, as mentioned at #6072.
> 
> So I manually reverted my working environment back to npm@5, but encountered more problems: `marked` requires higher version of node, rather than the ancient version of node@10 shipped with npm@5, etc. And also, it's obviously not best practice to always revert back to npm@5 to just please the lockfile version 1, right?
> 
> So getting a bit puzzled and wondering what is the best practice to add dependency to `/src`?
> 
> Thanks!

You could either downgrade to npm@6 or use the develop version with the latest pnpm.
